sdk/trace/provider.go

```diff
@@ -255,6 +256,7 @@ func (p *TracerProvider) Shutdown(ctx context.Context) error {
 			}
 		}
 	}
+	p.spanProcessors.Store(spanProcessorStates{})
```

Comment:
Do we also need to ensure new span processors cannot be registered?

Reply:
Shutdown is able to prevent future registered span processors